### DISCUSSION TITLE
When I set num_works> 0, there is a error   Producer process has been terminated before all shared CUDA tensors released

### DISCUSSION THREAD

**Discussion by Struggle-Forever** [Lightning-AI/pytorch-lightning#15322] [DDP / multi-GPU / multi-node]
![image](https://user-images.githubusercontent.com/29114869/197918436-af87214b-cc26-4183-8a71-2f0c9d89e007.png)
![image](https://user-images.githubusercontent.com/29114869/197918464-0c5a7bea-7fcb-4603-9953-987a2825561c.png)

**Comment by akihironitta** [comment]
`lr_lambda` needs to be picklable in your code (or in the library you use) in some strategy:
- https://pytorch-lightning.readthedocs.io/en/1.7.7/accelerators/gpu_faq.html?highlight=pickle#i-m-getting-errors-related-to-pickling-what-do-i-do
- https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled

**Reply by Struggle-Forever** [reply]
Thank you! I love pytorch-lightning!

**Comment by danielajisafe** [comment]
In my case, this issue was because my tensors were already placed in cuda within the getitem() and I also had ```torch.set_default_tensor_type('torch.cuda.FloatTensor')``` before the general main function in the code which technically was calling cuda on cuda tensor. So I either dropped one of them or put everything on CPU as my task did not require a GPU.
